lib/plugins/wallet-scoring/scorechain/scorechain.js

```diff
@@ -0,0 +1,74 @@
+const axios = require('axios')
+const _ = require('lodash/fp')
+
+
+const NAME = 'Scorechain'
+const SUPPORTED_COINS = {
+  BTC: 'BITCOIN',
+  ETH: 'ETHEREUM',
+  USDT: 'ETHEREUM',
+  BCH: 'BITCOINCASH',
+  LTC: 'LITECOIN',
+  DASH: 'DASH',
+  TRX: 'TRON',
+  USDT_TRON: 'TRON'
+}
+
+const TYPE = {
+  TRANSACTION: 'TRANSACTION',
+  ADDRESS: 'ADDRESS'
+}
+
+function rate (account, objectType, cryptoCode, objectId) {
+  if (_.isNil(account) || !account.enabled || !Object.keys(SUPPORTED_COINS).includes(cryptoCode)) return Promise.resolve(null)
+
+  const threshold = account.scoreThreshold
+  const payload = {
+    analysisType: 'ASSIGNED',
+    objectType,
+    objectId,
+    blockchain: SUPPORTED_COINS[cryptoCode],
+    coin: "ALL"
+  }
+
+  return axios.post(`https://api.scorechain.com/v1/scoringAnalysis`, payload, { headers: { 'X-API-KEY': account.apiKey }
+  })
+    .then(res => {
+      const resScore = res.data?.analysis?.assigned?.result?.score
+      if (!resScore) throw new Error('Failed to get score from Scorechain API')
+
+      // normalize score to 0-10 where 0 is the highest risk
+      // use 101 instead of 100 to avoid division by zero
+      return { score: (101 - resScore) / 10 - 0.1, isValid: resScore >= threshold }
```

Comment:
Why is 0 the highest risk? I seem to remember 0 being the lowest and 10 the highest with Ciphertrace, and that being colorcoded as green-yellow-red accordingly in l-a; however I don't see changes related to that in this PR.

Also, there's no division by zero possible..? Methinks `score: resScore/10` is fine (or `score: (100-resScore)/10` for the inverse order).

Reply:
Idk why I decided to prevent division by zero, maybe I had it the other way around on my mind.

The other one is a mistake on the comment, this normalizes to 0 being lowest risk, which was previously used.